plugins/input/netping/netping.go

```diff
@@ -155,12 +183,79 @@ func (m *NetPing) Description() string {
 	return "a icmp-ping/tcp-ping plugin for logtail"
 }
 
+func (m *NetPing) evaluteDNSResolve(host string) {
+	success := true
+	start := time.Now()
+	ips, err := net.LookupIP(host)
+	rt := time.Since(start)
+	if err != nil {
+		success = false
+		m.resolveHostMap.Store(host, "")
+	} else {
+		var n int64
+		nBig, err := rand.Int(rand.Reader, big.NewInt(int64(len(ips))))
+		if err == nil {
+			n = nBig.Int64()
+		}
+		m.resolveHostMap.Store(host, ips[n].String())
+	}
+
+	var label helper.KeyValues
+	label.Append("target", host)
+	label.Append("src", m.hostname)
+	label.Append("src_host", m.hostname)
+
+	m.resolveChannel <- &ResolveResult{
+		Success: success,
+		RTMs:    float64(rt.Milliseconds()),
+		Label:   label.String(),
+	}
+}
+
+func (m *NetPing) getRealTarget(target string) string {
+	realTarget := ""
+	v, exists := m.resolveHostMap.Load(target)
+	if exists {
```

Comment:
if else？

Reply:
else will  be covered

	if realTarget == "" {
		realTarget = target
	}